yarn-project/prover-client/src/orchestrator/orchestrator.ts

```diff
@@ -210,7 +209,6 @@ export class ProvingOrchestrator {
    * Cancel any further proving of the block
    */
   public cancelBlock() {
-    this.queue.cancelAll();
```

Comment:
I'm relying on the proving state being cancelled which automatically rejects every job result. I think this should be using `AbortSignal`:

In `deferredProving` create an `AbortController` and give the request the `AbortSignal`
```
const safeJob = async () => {
  const controller = new AbortController();
  this.inFlightRequests.push(controller);

  const result = await request(controller.signal);
  // finished, remove controller
  this.inFlightRequests.splice(this.inFlightRequests.indexOf(controller), 1);

  await callback(result);
};
```

And in `cancelBlock`:

```
public cancelBlock() {
  this.inFlightRequests.forEach(controller => controller.abort());
  this.provingState.cancel();
}
```

The event-driven nature of this would allow provers to cancel in flight proving requests (e.g. the native provers could `ctrl-c` the child process they spawned)